sonar-core/build.gradle

```diff
@@ -1,3 +1,7 @@
+plugins {
```

Comment:
why was the java-library plugin added  here?

Reply:
@runningcode I need to make some of the dependencies `api`

Reply:
How was any java functionality added here before? Is there an `allprojects` block adding the `java` plugin or something like that?

Reply:
Yeah, I think it would make sense to change this to `java-library`: https://github.com/SonarSource/sonarqube/blob/master/build.gradle#L163

It's a bad practice to use the allprojects block like that but let's not restrcuture their build right now.

Reply:
@runningcode you are right, I updated the plugin there and removed it from everywhere else